packages/medusa/src/services/discount.ts

```diff
@@ -759,6 +773,35 @@ class DiscountService extends TransactionBaseService {
     return !!limit && count >= limit
   }
 
+  async hasCustomerReachedLimit(
+    discount: Discount,
+    customer: Customer
+  ): Promise<boolean> {
+    // If customer is empty on the cart, we will allow it
+    if (!customer) {
+      return false
+    }
+
+    const manager = this.manager_
```

Comment:
todo: Use active manager
```suggestion
    const manager = this.activeManager_
```